test/repl.jl

```diff
@@ -348,7 +348,7 @@ begin
     LineEdit.accept_result(s, histp)
     @test LineEdit.mode(s) == repl_mode
     @test buffercontents(LineEdit.buffer(s)) == "shell"
-    @test position(LineEdit.buffer(s)) == 4
+    @test position(LineEdit.buffer(s)) == 2
```

Comment:
I've spent a day and a half trying to track down why this changed, with no success. The LineEdit code is totally impossible to debug – it's a poster child of mutable state and confusing dispatch. Using Gallium on it is impossible since the code I'm debugging is used extensively by Gallium itself (setting a breakpoint in LineEdit while using Gallium does not go well). The way the tests are written makes it impossible to reproduce the issue without running the tests – and refactoring the tests so that each set of tests is self-contained is beyond the scope of what I'm willing to do for this issue. Unless someone else (@Keno?) is willing to look at this, rather than spend any more of my time on this, I propose merging this change and opening an issue about figuring out why this changed later.